### PR TITLE
Added: Implementation of POST login and logout

### PR DESCRIPTION
**Pull request by simon-spier0**
Hi,

I added support for POST in login and logout.
#308, #280

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -100,12 +100,30 @@ This will add these paths to Django:
...
+
 .. code-block:: html
 
     <a href="{% url 'django_auth_adfs:logout' %}">Logout</a>
     <a href="{% url 'django_auth_adfs:login' %}">Login</a>
     <a href="{% url 'django_auth_adfs:login-no-sso' %}">Login (no SSO)</a>
 
+- POST (recommended):
+
+.. code-block:: html
```
Comment (tim-schilling) [inline README.rst:113]:
```suggestion
.. code-block:: html+django
```

File: README.rst [hunk 2]
```diff
@@ -100,12 +100,30 @@ This will add these paths to Django:
...
+        <button type="submit">Logout</button>
+    </form>
+    <form method="post" action="{% url 'django_auth_adfs:login' %}">{% csrf_token %}
+        <input type="hidden" name="next" value="{{ next }}">
+        <button type="submit">Login</button>
+    </form>
+    <form method="post" action="{% url 'django_auth_adfs:login-no-sso' %}">{% csrf_token %}
+        <input type="hidden" name="next" value="{{ next }}">
+        <button type="submit">Login (no SSO)</button>
+    </form>
```
Comment (tim-schilling) [inline README.rst:125]: What do you think about adding the following form example? It's something I had to do for another project and it's limited to requiring javascript.

```django
{# Requires javascript, but avoids the button element #}
<form method="post" action="{% url 'django_auth_adfs:logout' %}">{% csrf_token %}
    <a href="#" onclick="this.closest('form').submit()">Logout</a>
</form>
```
```

**Review by tim-schilling — CHANGES_REQUESTED** [review]
I'm all for changing the logout to be a POST. If you want to update this PR to just be that change we can get it merged.

Let's discuss login flows in #308. I'm sorry for the misunderstanding.

File: README.rst (hunk 2 above)
Comment (simon-spier0) [reply README.rst:125]: Thanks for the suggestion! I see the advantages of using the link approach, especially for styling purposes. However, I’m leaning towards using the standard <button> element for a few reasons:

- **CSP Compatibility**: Inline JavaScript might conflict with strict Content Security Policy (CSP) settings.

- **URL Issue**: The **href="#"** adds **#** to the URL, which could create unnecessary entries in the browser history and potentially confuse users.

That said, I’m happy to include both methods in the documentation if you’re okay with it.

**Comment by simon-spier0** [comment]
Just to confirm, would you like me to switch the logout to use only POST and remove the current GET method, or should I add POST support and keep GET as it is?

File: README.rst (hunk 2 above)
Comment (tim-schilling) [reply README.rst:125]: Nope, that's good enough for me. Thanks for the feedback!

**Comment by tim-schilling** [comment]
I think we need to continue to support logouts with GET until django 4.2 is no longer supported. That [matches Django's policy.](https://docs.djangoproject.com/en/4.2/topics/auth/default/#django.contrib.auth.views.LogoutView)

File: docs/install.rst [hunk 3]
```diff
@@ -104,8 +104,26 @@ This will add these paths to Django:
 
 You can use them like this in your django templates:
```
Comment (tim-schilling) [inline docs/install.rst:105]:
```suggestion
Below is sample Django template code to use these paths depending if
you'd like to use GET or POST requests. Logging out was deprecated in
`Django 4.1 <https://docs.djangoproject.com/en/5.1/releases/4.1/#features-deprecated-in-4-1>`_.
```

File: README.rst [hunk 4]
```diff
@@ -100,12 +100,30 @@ This will add these paths to Django:
 
 You can use them like this in your django templates:
```
Comment (tim-schilling) [inline README.rst:101]:
```suggestion
```suggestion
Below is sample Django template code to use these paths depending if
you'd like to use GET or POST requests. Logging out was deprecated in
`Django 4.1 <https://docs.djangoproject.com/en/5.1/releases/4.1/#features-deprecated-in-4-1>`_.
```
```

File: README.rst [hunk 5]
```diff
@@ -100,12 +100,30 @@ This will add these paths to Django:
...
 
+- GET:
```
Comment (tim-schilling) [inline README.rst:103]:
```suggestion
- Using GET requests:
```

File: README.rst [hunk 6]
```diff
@@ -100,12 +100,30 @@ This will add these paths to Django:
...
+
 .. code-block:: html
 
     <a href="{% url 'django_auth_adfs:logout' %}">Logout</a>
     <a href="{% url 'django_auth_adfs:login' %}">Login</a>
     <a href="{% url 'django_auth_adfs:login-no-sso' %}">Login (no SSO)</a>
 
+- POST:
```
Comment (tim-schilling) [inline README.rst:111]:
```suggestion
- Using POST requests:
```

**Review by tim-schilling — APPROVED** [review]
I'm in favor of this change. I'd like to include the additional docs changes if you're okay with that @simon-spier0 and @JonasKs

File: README.rst [hunk 7]
```diff
@@ -98,14 +98,35 @@ This will add these paths to Django:
 * ``/oauth2/callback`` where ADFS redirects back to after login. So make sure you set the redirect URI on ADFS to this.
 * ``/oauth2/logout`` which logs out the user from both Django and ADFS.
 
-You can use them like this in your django templates:
+```suggestion
```
Comment (JonasKs) [inline README.rst:101]: Will this render properly?

**Review by JonasKs — APPROVED** [review]

File: README.rst (hunk 7 above)
Comment (tim-schilling) [reply README.rst:101]: Ack, no. That is a github feature. I'll clean this up

**Comment by tim-schilling** [comment]
Thank you @simon-spier0!

**Pull request merged**